### ISSUE TITLE
Depends on nix-prefetch-git implicitly

### ISSUE THREAD

**Issue by workflow** [tenx-tech/cargo2nix#28]
Current `cargo2nix` has an implicit runtime dependency on `nix-prefetch-git` for workspaces that need to pull in crates via git at least.

I would add it as `propagatedBuildInput`, but cross-compilation for `nix-prefetch-git` on `musl64` is currently failing with some perl fu:

```
updateAutotoolsGnuConfigScriptsPhase                                                                             │Attempt to reload Scalar/Util.pm aborted.
patching ./t/write_default_maniskip.t...                                                                         │Compilation failed in require at /nix/store/kvadi9vl2alx6jy8hfbxr7ldccj184sc-perl-5.30.0-x86_64-unknown-linux-mu
Updating Autotools / GNU config script to a newer upstream version: ./build-aux/config.sub                       │sl/lib/perl5/5.30.0/x86_64-linux/Data/Dumper.pm line 304.
Updating Autotools / GNU config script to a newer upstream version: ./build-aux/config.guess                     │Couldn't run Build.PL: No such file or directory at lib/Module/Build/Compat.pm line 341.
patching ./t/PL_files.t...                                                                                       │cp lib/Test/Deep/Ref.pm blib/lib/Test/Deep/Ref.pm
configuring                                                                                                      │cp lib/Test/Deep/ArrayLength.pm blib/lib/Test/Deep/ArrayLength.pm
patching ./t/test_types.t...                                                                                     │cp lib/Test/Deep.pm blib/lib/Test/Deep.pm
patching ./t/files.t...                                                                                          │cp lib/Test/Deep/Boolean.pm blib/lib/Test/Deep/Boolean.pm
checking for x86_64-unknown-linux-musl-gcc... x86_64-unknown-linux-musl-gcc                                      │cp lib/Test/Deep/ArrayLengthOnly.pm blib/lib/Test/Deep/ArrayLengthOnly.pm
patching ./t/install.t...                                                                                        │cp lib/Test/Deep/Cache/Simple.pm blib/lib/Test/Deep/Cache/Simple.pm
Can't load module IO, dynamic loading not available in this perl.                                                │cp lib/Test/Deep/RefType.pm blib/lib/Test/Deep/RefType.pm
  (You may need to build a new perl executable which either supports                                             │cp lib/Test/Deep/HashKeysOnly.pm blib/lib/Test/Deep/HashKeysOnly.pm
  dynamic loading or has the IO module statically linked into it.)                                               │cp lib/Test/Deep/Array.pm blib/lib/Test/Deep/Array.pm
 at /nix/store/9jbxnmc146sh19x4ymkikyyava666m2a-perl-5.28.2-x86_64-unknown-linux-musl/lib/perl5/5.28.2/x86_64-lin│cp lib/Test/Deep/HashElements.pm blib/lib/Test/Deep/HashElements.pm
ux/IO/Handle.pm line 268.                                                                                        │cp lib/Test/Deep/ListMethods.pm blib/lib/Test/Deep/ListMethods.pm
Compilation failed in require at /nix/store/9jbxnmc146sh19x4ymkikyyava666m2a-perl-5.28.2-x86_64-unknown-linux-mus│cp lib/Test/Deep/RegexpRef.pm blib/lib/Test/Deep/RegexpRef.pm
l/lib/perl5/5.28.2/x86_64-linux/IO/Handle.pm line 268.                                                           │cp lib/Test/Deep/Stack.pm blib/lib/Test/Deep/Stack.pm
BEGIN failed--compilation aborted at /nix/store/9jbxnmc146sh19x4ymkikyyava666m2a-perl-5.28.2-x86_64-unknown-linux│cp lib/Test/Deep/Shallow.pm blib/lib/Test/Deep/Shallow.pm
-musl/lib/perl5/5.28.2/x86_64-linux/IO/Handle.pm line 268.                                                       │cp lib/Test/Deep/Isa.pm blib/lib/Test/Deep/Isa.pm
Compilation failed in require at /nix/store/9jbxnmc146sh19x4ymkikyyava666m2a-perl-5.28.2-x86_64-unknown-linux-mus│cp lib/Test/Deep/RegexpMatches.pm blib/lib/Test/Deep/RegexpMatches.pm
l/lib/perl5/5.28.2/x86_64-linux/IO/Seekable.pm line 100.                                                         │cp lib/Test/Deep/HashEach.pm blib/lib/Test/Deep/HashEach.pm
BEGIN failed--compilation aborted at /nix/store/9jbxnmc146sh19x4ymkikyyava666m2a-perl-5.28.2-x86_64-unknown-linux│cp lib/Test/Deep/MM.pm blib/lib/Test/Deep/MM.pm
-musl/lib/perl5/5.28.2/x86_64-linux/IO/Seekable.pm line 100.                                                     │x86_64-unknown-linux-musl-strip is /nix/store/gkq6rh8nj8i7prh4c9rawhwrk5g20cfx-x86_64-unknown-linux-musl-binutil
Compilation failed in require at /nix/store/9jbxnmc146sh19x4ymkikyyava666m2a-perl-5.28.2-x86_64-unknown-linux-mus│s-2.31.1/bin/x86_64-unknown-linux-musl-strip
l/lib/perl5/5.28.2/x86_64-linux/IO/File.pm line 132.                                                             │Removing empty /nix/store/q76nwhzd6v9b5j8kf2kchqn4b5yzsd93-perl5.30.0-Test-RequiresInternet-0.05-x86_64-unknown-
BEGIN failed--compilation aborted at /nix/store/9jbxnmc146sh19x4ymkikyyava666m2a-perl-5.28.2-x86_64-unknown-linux│linux-musl/share/man/ and (possibly) its parents
-musl/lib/perl5/5.28.2/x86_64-linux/IO/File.pm line 132.                                                         │patching script interpreter paths in /nix/store/ivw23rb4hk0mfl6smd37gqs7ak0dlwb0-perl5.30.0-IO-HTML-1.001-x86_64
Compilation failed in require at Makefile.PL line 5.                                                             │-unknown-linux-musl-devdoc
BEGIN failed--compilation aborted at Makefile.PL line 5.                                                         │cp lib/Test/Deep/All.pm blib/lib/Test/Deep/All.pm
configure flags: --disable-dependency-tracking --prefix=/nix/store/wbn67ycs6vnw2qddqh00jxfhwwysafzi-cpio-2.12-x86│cp lib/Test/Deep/RegexpVersion.pm blib/lib/Test/Deep/RegexpVersion.pm
_64-unknown-linux-musl --build=x86_64-unknown-linux-gnu --host=x86_64-unknown-linux-musl                         │builder for '/nix/store/khsym56j66s32ki1sgfqvpa0v36fcf5l-perl5.30.0-Module-Build-0.4229-x86_64-unknown-linux-mus
patching ./t/extend.t...                                                                                         │l.drv' failed with exit code 2
builder for '/nix/store/9kyibxdmp4nw8a9glvz79vg69ycqrhih-perl5.28.2-FCGI-0.78-x86_64-unknown-linux-musl.drv' fail│cannot build derivation '/nix/store/cb67anzy216wrqq5jl49azai6wad7pi4-perl5.30.0-HTML-TagCloud-0.38-x86_64-unknow
ed with exit code 255                                                                                            │n-linux-musl.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/z4j738hxcnpmpa4m65k9xkz1bwjw5237-git-2.22.0-x86_64-unknown-linux-musl.drv': 1│cannot build derivation '/nix/store/n2986pg9s46mkhm866rcv7211bd99c85-git-2.23.0-x86_64-unknown-linux-musl.drv': 
 dependencies couldn't be built                                                                                  │1 dependencies couldn't be built
cannot build derivation '/nix/store/qvyg35knid1fb93iyjj0jl742jamyhkk-nix-prefetch-git-x86_64-unknown-linux-musl.d│cannot build derivation '/nix/store/g3q6c0wiiq4df7p0bz1wx86qskg18csa-nix-prefetch-git-x86_64-unknown-linux-musl.
rv': 1 dependencies couldn't be built                                                                            │drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/0iwsy3wlkd3ccaninza76a0m7kdbvjyp-crate-cargo2nix-0.3.0-x86_64-unknown-linux-m│error: build of '/nix/store/g3q6c0wiiq4df7p0bz1wx86qskg18csa-nix-prefetch-git-x86_64-unknown-linux-musl.drv' fai
usl.drv': 1 dependencies couldn't be built                                                                       │led
error: build of '/nix/store/0iwsy3wlkd3ccaninza76a0m7kdbvjyp-crate-cargo2nix-0.3.0-x86_64-unknown-linux-musl.drv'│ !  ~/c/t/cumbersell   std-nix-setup *~$…                               4m  Wed 30 Oct 2019 12:20:24 PM +08
 failed    
```

When I run `nix-shell -p pkgsCross.musl64.nix-prefetch-git` with no funky perl overrides on a fairly recent `nixpkgs`, I get this:

```
x86_64-unknown-linux-musl-strip is /nix/store/gkq6rh8nj8i7prh4c9rawhwrk5g20cfx-x86_64-unknown-linux-musl-binutils-2.31.1/bin/x86_64-unknown-linux-musl-strip
cp lib/Test/Deep/NoTest.pm blib/lib/Test/Deep/NoTest.pm
cp lib/Test/Deep/Hash.pm blib/lib/Test/Deep/Hash.pm
cp lib/Test/Deep/Any.pm blib/lib/Test/Deep/Any.pm
cp lib/Test/Deep/Blessed.pm blib/lib/Test/Deep/Blessed.pm
cp lib/Test/Deep/Regexp.pm blib/lib/Test/Deep/Regexp.pm
cp lib/Test/Deep/Obj.pm blib/lib/Test/Deep/Obj.pm
cp lib/Test/Deep/RegexpRefOnly.pm blib/lib/Test/Deep/RegexpRefOnly.pm
cp lib/Test/Deep/Methods.pm blib/lib/Test/Deep/Methods.pm
cp lib/Test/Deep/RegexpOnly.pm blib/lib/Test/Deep/RegexpOnly.pm
cp lib/Test/Deep/None.pm blib/lib/Test/Deep/None.pm
cp lib/Test/Deep/Ignore.pm blib/lib/Test/Deep/Ignore.pm
cp lib/Test/Deep/Set.pm blib/lib/Test/Deep/Set.pm
cp lib/Test/Deep/Number.pm blib/lib/Test/Deep/Number.pm
Removing empty /nix/store/inzrdszvd9n7lrl9ja4vp73cwiwsss4n-perl5.30.0-Test-NoWarnings-1.04-x86_64-unknown-linux-musl/share/man/ and (possibly) its parents
patching script interpreter paths in /nix/store/2ndkbbmigyb4b3nd837ac9j2smr081c9-perl5.30.0-Test-Needs-0.002006-x86_64-unknown-linux-musl-devdoc
checking for references to /tmp/nix-build-perl5.30.0-FCGI-ProcManager-0.28-x86_64-unknown-linux-musl.drv-0/ in /nix/store/5ijqhf0yy4jrzwv75c11z88pnr999fq6-perl5.30.0-FCGI-ProcManager-0.28-x86_64-unknown-linux-musl-devdoc...
Attempt to reload Scalar/Util.pm aborted.
Compilation failed in require at /nix/store/kvadi9vl2alx6jy8hfbxr7ldccj184sc-perl-5.30.0-x86_64-unknown-linux-musl/lib/perl5/5.30.0/x86_64-linux/Data/Dumper.pm line 304.
Couldn't run Build.PL: No such file or directory at lib/Module/Build/Compat.pm line 341.
cp lib/Test/Deep/Ref.pm blib/lib/Test/Deep/Ref.pm
cp lib/Test/Deep/ArrayLength.pm blib/lib/Test/Deep/ArrayLength.pm
cp lib/Test/Deep.pm blib/lib/Test/Deep.pm
cp lib/Test/Deep/Boolean.pm blib/lib/Test/Deep/Boolean.pm
cp lib/Test/Deep/ArrayLengthOnly.pm blib/lib/Test/Deep/ArrayLengthOnly.pm
cp lib/Test/Deep/Cache/Simple.pm blib/lib/Test/Deep/Cache/Simple.pm
cp lib/Test/Deep/RefType.pm blib/lib/Test/Deep/RefType.pm
cp lib/Test/Deep/HashKeysOnly.pm blib/lib/Test/Deep/HashKeysOnly.pm
cp lib/Test/Deep/Array.pm blib/lib/Test/Deep/Array.pm
cp lib/Test/Deep/HashElements.pm blib/lib/Test/Deep/HashElements.pm
cp lib/Test/Deep/ListMethods.pm blib/lib/Test/Deep/ListMethods.pm
cp lib/Test/Deep/RegexpRef.pm blib/lib/Test/Deep/RegexpRef.pm
cp lib/Test/Deep/Stack.pm blib/lib/Test/Deep/Stack.pm
cp lib/Test/Deep/Shallow.pm blib/lib/Test/Deep/Shallow.pm
cp lib/Test/Deep/Isa.pm blib/lib/Test/Deep/Isa.pm
cp lib/Test/Deep/RegexpMatches.pm blib/lib/Test/Deep/RegexpMatches.pm
cp lib/Test/Deep/HashEach.pm blib/lib/Test/Deep/HashEach.pm
cp lib/Test/Deep/MM.pm blib/lib/Test/Deep/MM.pm
x86_64-unknown-linux-musl-strip is /nix/store/gkq6rh8nj8i7prh4c9rawhwrk5g20cfx-x86_64-unknown-linux-musl-binutils-2.31.1/bin/x86_64-unknown-linux-musl-strip
Removing empty /nix/store/q76nwhzd6v9b5j8kf2kchqn4b5yzsd93-perl5.30.0-Test-RequiresInternet-0.05-x86_64-unknown-linux-musl/share/man/ and (possibly) its parents
patching script interpreter paths in /nix/store/ivw23rb4hk0mfl6smd37gqs7ak0dlwb0-perl5.30.0-IO-HTML-1.001-x86_64-unknown-linux-musl-devdoc
cp lib/Test/Deep/All.pm blib/lib/Test/Deep/All.pm
cp lib/Test/Deep/RegexpVersion.pm blib/lib/Test/Deep/RegexpVersion.pm
builder for '/nix/store/khsym56j66s32ki1sgfqvpa0v36fcf5l-perl5.30.0-Module-Build-0.4229-x86_64-unknown-linux-musl.drv' failed with exit code 2
cannot build derivation '/nix/store/cb67anzy216wrqq5jl49azai6wad7pi4-perl5.30.0-HTML-TagCloud-0.38-x86_64-unknown-linux-musl.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/n2986pg9s46mkhm866rcv7211bd99c85-git-2.23.0-x86_64-unknown-linux-musl.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/g3q6c0wiiq4df7p0bz1wx86qskg18csa-nix-prefetch-git-x86_64-unknown-linux-musl.drv': 1 dependencies couldn't be built
error: build of '/nix/store/g3q6c0wiiq4df7p0bz1wx86qskg18csa-nix-prefetch-git-x86_64-unknown-linux-musl.drv' failed
```

**Comment by dingxiangfei2009** [comment]
Switch to `propagatedNativeBuildInputs`?

**Comment by workflow** [comment]
Hello hello @dingxiangfei2009 :wave: :)
I think this has been solved by one of @trha 's latest PRs, lemme check when I come home :)

**Comment by workflow** [comment]
Yep, looks like it's fixed in 0.5.0! :taco:
